tools/kagami/src/swarm.rs

```diff
@@ -0,0 +1,1300 @@
+const GIT_ORIGIN: &str = "https://github.com/hyperledger/iroha.git";
+/// Config directory that is generated in the output directory
+const DIR_CONFIG: &str = "config";
+/// Config directory inside of the docker image
+const DIR_CONFIG_IN_DOCKER: &str = "/config";
+const DIR_CLONE: &str = "iroha-cloned";
+const FILE_VALIDATOR: &str = "validator.wasm";
+const FILE_CONFIG: &str = "config.json";
+const FILE_GENESIS: &str = "genesis.json";
+const FILE_COMPOSE: &str = "docker-compose.yml";
+const DIR_FORCE_SUGGESTION: &str =
+    "You can pass `--outdir-force` flag to remove the directory without prompting";
+const GENESIS_KEYPAIR_SEED: &[u8; 7] = b"genesis";
+
+#[derive(ClapArgs, Debug)]
+pub struct Args {
+    #[command(flatten)]
+    source: ImageSourceArgs,
+    /// How many peers to generate within the docker-compose.
+    #[arg(long, short)]
+    peers: NonZeroU16,
+    /// Target directory where to place generated files.
+    ///
+    /// If the directory is not empty, Kagami will prompt it's re-creation. If the TTY is not
+    /// interactive, Kagami will stop execution with non-zero exit code. In order to re-create
+    /// the directory anyway, pass `--outdir-force` flag.
+    #[arg(long)]
+    outdir: PathBuf,
+    /// Re-create the target directory if it already exists.
+    #[arg(long)]
+    outdir_force: bool,
+    /// Do not create default configuration in the `<outdir>/config` directory.
+    ///
+    /// Default `config.json`, `genesis.json` and `validator.wasm` are generated and put into
+    /// the `<outdir>/config` directory. That directory is specified in the `volumes` field
+    /// of the Docker Compose file.
+    ///
+    /// Setting this flag prevents copying of default configuration files into the output folder. The `config` directory will still be
+    /// created, but the necessary configuration should be put there by the user manually.
+    #[arg(long)]
+    no_default_configuration: bool,
+    /// Might be useful for deterministic key generation.
+    ///
+    /// It could be any string. Its UTF-8 bytes will be used as a seed.
+    #[arg(long, short)]
+    seed: Option<String>,
+}
+
+impl Args {
+    pub fn run(self) -> Outcome {
+        let ui = UserInterface::new();
+
+        let prepare_dir_strategy = if self.outdir_force {
+            PrepareDirectoryStrategy::ForceRecreate
+        } else {
+            PrepareDirectoryStrategy::Prompt
+        };
+        let source = ImageSource::from(self.source);
+        let target_dir = TargetDirectory::new(AbsolutePath::absolutize(self.outdir)?);
+
+        if let EarlyEnding::Halt = target_dir
+            .prepare(&prepare_dir_strategy, &ui)
+            .wrap_err("failed to prepare directory")?
+        {
+            return Ok(());
+        }
+
+        let config_dir = AbsolutePath::absolutize(target_dir.path.join(DIR_CONFIG))?;
+
+        let source = source
+            .resolve(&target_dir, &ui)
+            .wrap_err("failed to resolve the source of image")?;
```

Comment:
```suggestion
            .wrap_err("Failed to resolve the source of image")?;
```
Here and in all subsequent messages as that's usually our style.